tokio-buf/src/util/stream.rs

```diff
@@ -33,3 +33,29 @@ where
         self.stream.poll()
     }
 }
+
+/// Converts a `BufStream` into a `Stream`.
+#[derive(Debug)]
+pub struct IntoStream<T> {
+    buf: T,
+}
+
+impl<T> IntoStream<T> {
+    /// Create a new `Stream` from the provided `BufStream`.
+    pub fn new(buf: T) -> Self {
```

Comment:
Can we have `get_ref`, `get_mut` and `into_inner` ?

Reply:
This is a good thought.

Reply:
done https://github.com/tokio-rs/tokio/pull/1048/commits/73c6a2e1975cea801619d8d7c468421553e9f6fd